src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -1456,6 +1459,10 @@ void GcsActorManager::RestartActor(const ActorID &actor_id,
   if (remaining_restarts != 0) {
     // num_restarts must be set before updating GCS, or num_restarts will be inconsistent
     // between memory cache and storage.
+    if (mutable_actor_table_data->preempted()) {
+      mutable_actor_table_data->set_num_restarts_due_to_node_preemption(
+          num_restarts_due_to_node_preemption + 1);
+    }
```

Comment:
what's the reason for adding a new field instead of not incrementing `num_restarts` when the node was pre-empted?

Reply:
I think it is getting more visibility for what is happening to an actor. But seems like most people don't like it, I just changed to not incrementing the num_restarts.